src/resources/dart-3-migration.md

```diff
@@ -0,0 +1,512 @@
+---
+title: Dart 3 migration guide
+description: Tips for how to migrate existing Dart code to be compatible with Dart 3.
+---
+
+Dart 3 is a new major release of Dart, that brings new core capabilities like
+records, patterns, and classes. Alongside these new capabilities the release
+also contains a number of changes, that may break existing code. The present
+document is intended to be a migration guide that may be helpful in resolving
+these.
+
+## Introduction
+
+### Unversioned vs versioned changes
+
+The potentially breaking changes listed below fall into one of two categories:
+
+* **Unversioned changes**: These are changes which are affect any Dart code,
+which is used after upgrading to a Dart 3 SDK. There is no way to "turn off"
+these changes.
+
+* **Versioned changes**: Versioned changes are different, 
+and only apply when the package/app has selected the 
+Dart 3.0 (or above) language version. 
+The language version [is derived](/guides/language/evolution#language-version-numbers)
+from the `sdk` lower-constraint in the `pubspec.yaml` file. An SDK constraint like
+this does *not* apply the Dart 3 versioned changes:
+    ```yaml
+    environment:
+      sdk: '>=2.14.0 <3.0.0'
+    ```
+    But an SDK constraint like this does:
+    ```yaml
+    environment:
+      sdk: '>=3.0.0 <4.0.0'
+    ```
+
+Note that to use the new Dart 3 features you have to update the language version
+to 3.0, and thus you'll also get the Dart 3 versioned changes at the same time.
```

Comment:
I feel like this should be the first thing under the "Introduction" heading, before this section.

```suggestion
To use the new Dart 3 features you have to update the language version
to 3.0. This gets you the Dart 3 versioned changes at the same time.
```